app/models/miq_server/worker_management/monitor/kubernetes.rb

```diff
@@ -15,31 +20,88 @@ def failed_deployments(restart_count = 5)
     current_pods.values.select { |h| h[:last_state_terminated] && h.fetch(:container_restarts, 0) > restart_count }.collect { |h| h[:label_name] }
   end
 
+  def sync_deployment_settings
+    return unless podified?
+
+    checked_deployments = Set.new
+    podified_miq_workers.each do |worker|
+      next if checked_deployments.include?(worker.worker_deployment_name)
+
+      if deployment_resource_constraints_changed?(worker)
+        _log.info("Constraints changed, patching deployment: [#{worker.worker_deployment_name}]")
+        worker.patch_deployment
+      end
+      checked_deployments << worker.worker_deployment_name
+    end
+  end
+
+  def podified_miq_workers
+    # Cockpit is a threaded worker in the orchestrator that spins off a process it monitors and isn't a pod worker.
+    miq_workers.where.not(:type => %w[MiqCockpitWsWorker])
+  end
+
+  def deployment_resource_constraints_changed?(worker)
+    return false unless Settings.server.worker_monitor.enforce_resource_constraints
+
+    container = current_deployments.fetch_path(worker.worker_deployment_name, :spec, :template, :spec, :containers).try(:first)
+    current_constraints = container.try(:fetch, :resources, nil) || {}
+    desired_constraints = worker.resource_constraints
+    constraints_changed?(current_constraints, desired_constraints)
+  end
+
+  def constraints_changed?(current, desired)
+    return true if current.blank? && desired.present?
+    return true if current.present? && desired.blank?
```

Comment:
```suggestion
    return true if current.blank? ^ desired.blank?
```

XOR FTW

Reply:
🤣   XOR.  While true, I don't think that's as readable for "normal" people.